src/promptflow/promptflow/executor/_tool_resolver.py

```diff
@@ -41,12 +39,11 @@ class ResolvedTool:
     init_args: dict
 
 
-class ToolResolver(ThreadLocalSingleton):
-    CONTEXT_VAR_NAME = "Resolver"
-    context_var = ContextVar(CONTEXT_VAR_NAME, default=None)
+class ToolResolver():
 
     def __init__(
-        self, working_dir: Path, connections: Optional[dict] = None, package_tool_keys: Optional[List[str]] = None
+        self, working_dir: Path, connections: Optional[dict] = None, package_tool_keys: Optional[List[str]] = None,
+        need_connections: bool = True
```

Comment:
Not a good practice to add such a parameter.

Reply:
Removed.